### PR TITLE
Soft failing

### LINKED ISSUE #97: Phases and backend: make error non-fatal
When a phase or a backend cannot handle some piece of code, it currently spans an error and fails hard.
For the user experience, generating code anyway would be much nicer.

On failure on an expression, we should output a `hax_failure ERROR` expression where `ERROR` is the string representation of the error, followed by a multi-line comment in which we can place the Rust-ish pretty-printed expression we could not translate.

On failure on items, we should just output that Rust-ish pretty print as comments instead of the item.

Anyway, an error should make the tool exit code be non-zero: an error is still a fatal error, the fact we still produce files is to be considered as mean of debugging.

## TODO list
 - [ ] Add facilities to span errors without throwing an exception. I guess the easier thing to do is to use mutation in ML.
 - [ ] Add primitive to make `print_rust` module pretty print expressions as well.

### PR DESCRIPTION
**Pull request by W95Psp**
Fixes #97

TODO:
 - [x] print the most-advanced Rustish AST of the chunk of code that yields an error

### CODE REVIEW

**Comment by W95Psp** [comment]
For now, I print errors and ast in strings in the backend. When we have #117, will output that as comments

**Review by franziskuskiefer — APPROVED** [review]
Thanks, one nit and a question.
Let's get this in.

File: cli/driver/src/exporter.rs [hunk 1]
```diff
@@ -245,49 +245,46 @@ impl Callbacks for ExtractionCallbacks {
...
-                                    session.note_without_error(format!("Writing file {:#?}", path));
-                                    std::fs::write(&path, file.contents).unwrap_or_else(|e| {
-                                        session.fatal(format!(
-                                            "Unable to write to file {:#?}. Error: {:#?}",
-                                            path, e
-                                        ))
-                                    })
-                                }
-                            }
+                    for d in output.diagnostics.clone() {
```
[the remark this answers is not in the document]
Comment (W95Psp) [reply cli/driver/src/exporter.rs]: Seems like plain `for d in output.diagnstocis` was fine as well ^^

**Pull request merged**